### PR TITLE
close connection if handshake fails

### PR DESCRIPTION
**Pull request by qcdll**
this is a bit hacky but verified to fix the `too many open files` issue

### CODE REVIEW

File: quarkchain/p2p/p2p_server.py [hunk 1]
```diff
@@ -158,10 +158,17 @@ def _get_discv5_topic(self) -> bytes:
             await self._receive_handshake(reader, writer)
         except expected_exceptions as e:
             self.logger.debug("Could not complete handshake: %s", e)
+            self.logger.error("Could not complete handshake: %s", e)
```
Comment (ninjaahhh) [inline quarkchain/p2p/p2p_server.py]: duplicate line

File: quarkchain/p2p/p2p_server.py [hunk 2]
```diff
@@ -158,10 +158,17 @@ def _get_discv5_topic(self) -> bytes:
...
+            reader.feed_eof()
+            writer.close()
         except OperationCancelled:
-            pass
+            self.logger.error("OperationCancelled")
+            reader.feed_eof()
+            writer.close()
         except Exception as e:
             self.logger.exception("Unexpected error handling handshake")
+            reader.feed_eof()
```
Comment (ninjaahhh) [inline quarkchain/p2p/p2p_server.py]: can use `finally` block

File: quarkchain/p2p/p2p_server.py (hunk 1 above)
Comment (qcdll) [reply quarkchain/p2p/p2p_server.py]: they are not duplicate, but I will remove the error log

File: quarkchain/p2p/p2p_server.py (hunk 2 above)
Comment (qcdll) [reply quarkchain/p2p/p2p_server.py]: finally will close writer regardless of exception??

Comment (ninjaahhh) [reply quarkchain/p2p/p2p_server.py]: yup

Comment (qcdll) [reply quarkchain/p2p/p2p_server.py]: that's not right, shall not close if no exception

Comment (ninjaahhh) [reply quarkchain/p2p/p2p_server.py]: ah i see. another way to save those duplicate lines would be return at line 158 and close after the try-catch block

**Comment by qcdll** [comment]
actually this was recording opened sockets during handshake, and if the handshake turns out to be unfruitful we close the sockets

at the point of performing handshake, they are not "peers" yet

**Pull request merged**